ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -53,6 +52,33 @@ module Json = struct
   let merge_maps m maps =
     List.fold_left (fun acc map -> StringMap.union choose_enum acc map) m maps
 
+  let rec get_fp_type ty =
+    match ty with
+    | SecretString | String ->
+        "String"
+    | Int ->
+        "Int"
+    | Float ->
+        "Float"
+    | Bool ->
+        "Bool"
+    | DateTime ->
+        "Time"
+    | Enum (name, _) ->
+        "Enum" ^ snake_to_camel name
+    | Set ty ->
+        get_fp_type ty ^ "Set"
+    | Map (ty1, ty2) ->
+        let fp_type1 = get_fp_type ty1 in
+        let fp_type2 = get_fp_type ty2 in
+        fp_type1 ^ "To" ^ fp_type2 ^ "Map"
```

Comment:
I think I do not see names as `XToYMap` in `test_data/methods.go`, can it be added if it is not tested yet?

Reply:
Yes, unit tests should be added for `get_fp_type `.